libr/debug/dreg.c

```diff
@@ -128,9 +129,32 @@ R_API int r_debug_reg_list(RDebug *dbg, int type, int size, int rad, const char
 
 	int itmidx = -1;
 	dbg->creg = NULL;
-	head = r_reg_get_list (dbg->reg, type);
-	if (!head) {
-		return false;
+	if (type == R_REG_TYPE_ALL) {
+		head = r_list_new ();
+		for (i = 0; i < R_REG_TYPE_LAST; i++) {
+			RList *reglist = r_reg_get_list (dbg->reg, i);
+			if (!reglist || r_list_length (reglist) == 0) {
+				r_list_free (reglist);
+				continue;
+			}
+			RList *tmp = r_list_clone (reglist);
+			if (!tmp) {
+				r_list_free (reglist);
```

Comment:
why this? I think you said that `r_reg_get_list` does not return a newly allocated list, so freeing it will cause issues I think.

Reply:
you are right. To output all register types, I decided to clone and join all the lists together, so I can keep the outputting part of the code the same. If we change `r_reg_get_list` API to always return a newly allocated list, we can make `r_reg_get_list` to directly return a RList of the R_REG_TYPE_LAST type instead of doing it in `r_debug_reg_list`, which can be reused elsewhere.

Reply:
I don't have a strong opinion. You can either change the API so that it directly returns `R_REG_TYPE_ALL` or just do the chaining here as you are already doing :) I'm personally fine with both approaches for now.

Reply:
Ok, great. I think I will do it this way for now.